pkg/controller/dynamicrequiremanifest_controller.go

```diff
@@ -0,0 +1,181 @@
+	c := &dynamicRequiredManifestController{
+		manifest:         manifest,
+		name:             name,
+		key:              key,
+		gvr:              gvr,
+		applyFunc:        defaultApplyFunc(dynamicClient),
+		managedFunc:      defaultManagedFunc(operatorClient),
+		shouldUpdateFunc: unstructuredShouldUpdateFunc(),
+		objectGetFunc:    resourceClient.Get,
+	}
+
+	return factory.New().WithSync(c.sync).WithSyncDegradedOnError(operatorClient).WithInformers(operatorClient.Informer(), resourceClient.Informer()).ToController(c.name, recorder)
+}
+
+func defaultApplyFunc(client dynamic.Interface) applyFunc {
+	return func(ctx context.Context, key types.NamespacedName, fieldManager string, force bool, gvr schema.GroupVersionResource, manifest []byte) error {
+		var resourceInterface dynamic.ResourceInterface = client.Resource(gvr)
+		if key.Namespace != "" {
+			resourceInterface = client.Resource(gvr).Namespace(key.Namespace)
+		}
+		_, err := resourceInterface.Patch(
+			ctx,
+			key.Name,
+			types.ApplyPatchType,
+			manifest,
+			metav1.PatchOptions{
+				Force:        ptr.To(force),
+				FieldManager: fieldManager,
+			})
+		return err
+	}
+}
+
+func defaultManagedFunc(oc *clients.OperatorClient) managedFunc {
+	return func() (bool, error) {
+		operatorSpec, _, _, err := oc.GetOperatorState()
+		if err != nil {
+			return false, err
+		}
+		return management.IsOperatorManaged(operatorSpec.ManagementState) && (operatorSpec.ManagementState != operatorv1.Removed || management.IsOperatorNotRemovable()), nil
+	}
+}
+
+func unstructuredShouldUpdateFunc() shouldUpdateFunc {
+	return func(manifest []byte, existing runtime.Object) (bool, error) {
+		if existing == nil {
+			return true, nil
+		}
+
+		existingUnstructured, ok := existing.(*unstructured.Unstructured)
+		if !ok {
+			return false, fmt.Errorf("expected existing to be of type *unstructured.Unstructured but was %T", existing)
+		}
+
+		// Decode takes in a target type that is not guaranteed to be populated. Since we want an *unstructured.Unstructured type from decoding the
+		// manifest bytes, we pass in &unstructured.Unstructured and assume that the returned object is indeed an *unstructured.Unstructured
+		// For more information on the Decoder interface, see https://pkg.go.dev/k8s.io/apimachinery@v0.31.1/pkg/runtime#Decoder
+		required, _, err := scheme.Codecs.UniversalDecoder().Decode(manifest, nil, &unstructured.Unstructured{})
+		if err != nil {
+			return false, fmt.Errorf("decoding manifest: %w", err)
+		}
+
+		// we want the inverse of deep derivative to dictate if we should update.
+		// i.e If it is not deep derivative, meaning the requirements are not met, we should update
+		return !equality.Semantic.DeepDerivative(required.(*unstructured.Unstructured).UnstructuredContent(), existingUnstructured), nil
+	}
+}
```

Comment:
I can certainly add some more, but I feel the behavior is pretty straightforward for most of them and would prefer not mocking out Kubernetes clients as much as possible, namely the `dynamic.Interface` and `clients.OperatorClient`. That being said we could require the input to be minimal interfaces explicitly for our use cases. I think the most reasonable to add unit testing for would be the `unstructuredShouldUpdateFunc` function because it doesn't actually perform any calls to Kube and would be much more straightforward to test.

I'm happy to make any updates you feel necessary in terms of testing. I know you mention this is non-blocking, but I think we might as well make any additions we feel are important, including unit tests, to this code now rather than later (because we may never do it later).